edge-tool/tests/test_cbor.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2023 Izuma Networks
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Pytest based tests for testing certificate conversion to CBOR."""
+
+import os
+import filecmp
+
+
+def test_cbor_conversion(tmpdir):
+    """Test conversion (success)"""
+    tmp_outfile = str(tmpdir.join("out.cbor"))
+    ret = os.system(
+        "./edge_tool.py convert-dev-cert "
+        "--development-certificate test_data/mbed_cloud_dev_credentials.c "
+        " --update-resource test_data/update_default_resources.c "
+        f"--cbor {tmp_outfile}"
+    )
+    assert ret == 0
+    tmp_cbortxt = str(tmpdir.join("out.cbor.txt"))
+    ret = os.system(
+        f"./edge_tool.py print-cbor --cbor {tmp_outfile} >{tmp_cbortxt}"
+    )
```

Comment:
Well, they actually server a purpose here. This compares the produced data from the credentials files is matching via the meeans of the print-cbor (I couldn't get the binary file comparison to work for some reason). You could argue the 2nd test later on is useless with this being in place, but - well, the hack went in.

Reply:
But that isn't what the filecmp is comparing - it is comparing the cbor files directly.

Reply:
The binary comparison is failing, because the certs are timestamped (validity period starts from now and ends in 1 year). This causes subtle changes in the binary file which makes direct comparison impossible.

Reply:
Hmmh. You are correct, indeed. Fixed now.

Reply:
Interestingly, I ran coverage on this and with these 6 cases we get 100% coverage.